clarity_ext/domain/common.py

```diff
@@ -33,6 +33,9 @@ def _eq_rec(self, a, b, cache=[]):
                 continue
             elif any(b[key] is i for i in cache):
                 continue
+            elif a[key].__class__.__name__ == "MagicMock" and a[key].__class__.__name__ == "MagicMock":
```

Comment:
I definitely see the point with this as a short term solution to a problem, but I think this might give a false positive for other tests so I think we might want to refactor this later. I imagine other devs that haven't read this code writing tests that check for equality and don't know about this detail - they might get passed tests for domain objects that really are missing some crucial properties.

Without having dug deep, I would consider mocking the equality check instead. Then the responsibility for this resides in the tests only.